tools/esi/esi-cosim-runner.py.in

```diff
@@ -195,30 +197,40 @@ class CosimTestRunner:
       if not self.exec:
         self.writeScript(port)
 
-      # Pycapnp complains if the PWD environment var doesn't match the
-      # actual CWD.
-      testEnv = os.environ.copy()
-      testEnv["PWD"] = os.getcwd()
-      testEnv["PYTHONPATH"] = testEnv[
-          "PYTHONPATH"] + f":{os.path.dirname(__file__)}"
-      # Run the test script.
-      if self.exec:
-        args = [str(port), self.schema] + self.exec_args.split(" ")
-        if self.file.endswith(".py"):
-          cmd = [sys.executable, "-u", self.file] + args
-        else:
-          cmd = [self.file] + args
+      if self.server_only:
+        # wait for user input to kill the server
+        input(
+            f"Running in server-only mode on port {port} - Press anything to kill the server..."
+        )
       else:
-        cmd = [sys.executable, "-u", "script.py"]
-      print("[INFO] Test run command: " + " ".join(cmd))
-      testProc = subprocess.run(cmd,
-                                stdout=testStdout,
-                                stderr=testStderr,
-                                cwd=os.getcwd(),
-                                env=testEnv)
-      if not self.interactive:
-        testStdout.close()
-        testStderr.close()
+        # Test mode
+        # Pycapnp complains if the PWD environment var doesn't match the
+        # actual CWD.
+        testEnv = os.environ.copy()
+        testEnv["PWD"] = os.getcwd()
+        testEnv["PYTHONPATH"] = testEnv[
+            "PYTHONPATH"] + f":{os.path.dirname(__file__)}"
+        # Run the test script.
+        if self.exec:
+          args = [f"localhost:{port}", self.schema] + self.exec_args.split(" ")
```

Comment:
There was a specific reason I didn't use localhost but I don't recall what it was...